File: Year 3/Android Java/Simple SMHI Warning App JavaVersion/GeoWarning/app/src/main/java/com/example/geowarning/model/GeometryTypeEnum.java
package com.example.geowarning.model;

public enum GeometryTypeEnum {
    POINT("Point"),
    POLYGON("Polygon"),
    MULTI_POLYGON("MultiPolygon"),
    LINE_STRING("LineString");

    private final String value;

    GeometryTypeEnum(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    public static GeometryTypeEnum fromString(String text) {
        for (GeometryTypeEnum type : GeometryTypeEnum.values()) {
            if (type.value.equalsIgnoreCase(text)) {
                return type;
            }
        }
        throw new IllegalArgumentException("Unknown geometry type: " + text);
    }
}

File: Year 3/React Native/Simple SMHI Weather Warning application/my-second-project/models/GeometryTypeEnum.ts
export enum GeometryTypeEnum {
    POINT = "Point",
    POLYGON = "Polygon",
    MULTI_POLYGON = "MultiPolygon",
    LINE_STRING = "LineString"
}

export function fromString(text: string): GeometryTypeEnum {
    const value = Object.values(GeometryTypeEnum).find(v => v.toLowerCase() === text.toLowerCase());
    if (!value) {
        throw new Error(`Unknown geometry type: ${text}`);
    }
    return value as GeometryTypeEnum;
}